### PR TITLE
Invalidate the file system cache on deletion

### PR DESCRIPTION
**Pull request by thommay**
@chef/client-core

### CODE REVIEW

File: lib/chef/chef_fs/file_system/repository/directory.rb [hunk 1]
```diff
@@ -149,6 +150,10 @@ def exists?
 
           protected
 
+          def write(data)
+            raise "called write on #{file_path}"
```
Comment (danielsdeleo) [inline lib/chef/chef_fs/file_system/repository/directory.rb]: Is there a more specific exception class we could use?

**Comment by danielsdeleo** [comment]
Is this fixing a real-world bug? If so, is there any reasonable way to get the repro case in as a test?

File: lib/chef/chef_fs/file_system/repository/directory.rb (hunk 1 above)
Comment (thommay) [inline lib/chef/chef_fs/file_system/repository/directory.rb]: oof, that was debug spew that shouldn't have stayed in. But now you mention it, I think it's reasonable to raise a `Chef::ChefFS::FileSystem::FileSystemError` if someone does try to write to a directory entry.

**Comment by thommay** [comment]
Running pedant against chef-zero without `--skip-policies` found this one.

**Comment by danielsdeleo** [comment]
Cool, is it possible to remove `--skip-policies` from the pedant config or is there more work to do there?

**Comment by thommay** [comment]
Yeah, that PR is coming shortly :)

**Comment by danielsdeleo** [comment]
👍

**Comment by thommay** [comment]
https://github.com/chef/chef-zero/pull/231 is the chef zero PR

**Pull request merged**